src/common/framework/file_loader.ts

```diff
@@ -22,8 +23,12 @@ export abstract class TestFileLoader {
   abstract listing(suite: string): Promise<TestSuiteListing>;
   protected abstract import(path: string): Promise<SpecFile>;
 
-  importSpecFile(suite: string, path: string[]): Promise<SpecFile> {
-    return this.import(`${suite}/${path.join('/')}.spec.js`);
+  async importSpecFile(suite: string, path: string[]): Promise<SpecFile> {
+    const specFile = await this.import(`${suite}/${path.join('/')}.spec.js`);
+    assert(specFile.description !== undefined, 'Test spec file missing description: ' + path);
+    assert(specFile.g !== undefined, 'Test spec file missing TestGroup definition: ' + path);
+    specFile.g.validate();
```

Comment:
I think(?) moving this here prevents checking these things during presubmit. IDK if we should do it conditionally inside crawl() or if we should add another presubmit step for it.

Reply:
I think we can do it conditionally and pass some argument to gen_listings to tell it to check or not.

Reply:
SGTM